File: delivery-java-project/api/src/main/java/org/delivery/api/common/api/Result.java
package org.delivery.api.common.api;

import org.delivery.api.common.error.ErrorCode;
import org.delivery.api.common.error.ErrorCodeInterface;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@Builder
@NoArgsConstructor
public class Result {
  private Integer resultCode;
  private String resultMessage;
  private String resultDescription;

  public static Result ok() {
    return Result.builder()
        .resultCode(ErrorCode.OK.getErrorCode())
        .resultMessage(ErrorCode.OK.getDescription())
        .resultDescription("The request was successful.")
        .build();
  }

  public static Result ERROR(ErrorCodeInterface errorCodeInterface) {
    return Result.builder()
        .resultCode(errorCodeInterface.getErrorCode())
        .resultMessage(errorCodeInterface.getDescription())
        .resultDescription("The request was unsuccessful.")
        .build();
  }

  public static Result ERROR(ErrorCodeInterface errorCodeInterface, Throwable tx) {
    return Result.builder()
        .resultCode(errorCodeInterface.getErrorCode())
        .resultMessage(errorCodeInterface.getDescription())
        .resultDescription(tx.getLocalizedMessage())
        .build();
  }

    public static Result ERROR(ErrorCodeInterface errorCodeInterface, String description) {
    return Result.builder()
        .resultCode(errorCodeInterface.getErrorCode())
        .resultMessage(errorCodeInterface.getDescription())
        .resultDescription(description)
        .build();
  }
}

File: delivery-java-project/common/src/main/kotlin/org/delivery/common/api/Result.kt
package org.delivery.common.api

import org.delivery.common.error.ErrorCode
import org.delivery.common.error.ErrorCodeInterface

class Result(
	val resultCode: Int,
	val resultMessage: String,
	val resultDescription: String
) {
	companion object {
		fun ok(): Result {
			return Result(ErrorCode.OK.getErrorCode(), ErrorCode.OK.getDescription(), "The request was successful.")
		}
		
		fun ERROR(errorCode: ErrorCodeInterface): Result {
			return Result(
				resultCode = errorCode.getErrorCode(),
				resultDescription = errorCode.getDescription(),
				resultMessage = "The request was failed."
			)
		}
		
		fun ERROR(errorCode: ErrorCodeInterface, throwable: Throwable): Result {
			return Result(
				resultCode = errorCode.getErrorCode(),
				resultDescription = throwable.localizedMessage,
				resultMessage = errorCode.getDescription()
			)
		}
		
		fun ERROR(errorCode: ErrorCodeInterface, description: String): Result {
			return Result(
				resultCode = errorCode.getErrorCode(),
				resultDescription = description,
				resultMessage = errorCode.getDescription()
			)
		}
	}
}